plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/MavenAppEngineStandardWizardPage.java

```diff
@@ -184,6 +186,20 @@ private void createAppEngineProjectDetailsArea(Composite container) {
     javaPackagePosition.horizontalSpan = 2;
     javaPackageField.setLayoutData(javaPackagePosition);
     javaPackageField.addModifyListener(pageValidator);
+    javaPackageField.addVerifyListener(new VerifyListener() {
+
+      @Override
+      public void verifyText(VerifyEvent event) {
+        // check if set programmatically
+        if (javaPackageProgrammaticUpdate) {
+          return;
+        }
+        String current = ((Text) event.widget).getText();
+        String newText =
+            current.substring(0, event.start) + event.text + current.substring(event.end);
+        shouldUpdatePackageName = newText.isEmpty();
```

Comment:
Why check if it's empty? If the user explicitly set the package name, including setting it to empty by deleting the auto-generated value, we should not overwrite that.